### PR TITLE
[Sprint 4] Create Page Container Profile Information

### PR DESCRIPTION
**Pull request by Tuanlvt**
UI Pesonal Information + UI Category Favorite	
Link redmine: https://dev.framgia.com/redmine/issues/135512

### CODE REVIEW

File: app/src/main/java/com/framgia/fbook/screen/menuprofile/MenuProfileModule.kt [hunk 1]
```diff
@@ -24,4 +25,11 @@ class MenuProfileModule(private val fragment: Fragment) {
   fun provideNavigator(): Navigator {
     return Navigator(fragment)
   }
+
+  @FragmentScope
+  @Provides
+  fun provideAdapter(): MenuProfileAdapter {
+    return MenuProfileAdapter(fragment.context,
+        (fragment.activity as FragmentActivity).supportFragmentManager)
```
Comment (HyperionsThs) [inline app/src/main/java/com/framgia/fbook/screen/menuprofile/MenuProfileModule.kt]: change name fragment -> mFragment

File: app/src/main/res/layout/fragment_menu_profile_container.xml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<FrameLayout
```
Comment (trainer-android-fr) [inline app/src/main/res/layout/fragment_menu_profile_container.xml]: WARNING: [UnusedResources] The resource `R.layout.fragment_menu_profile_container` appears to be unused
 Unused resources make applications larger and slow down builds.

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -36,4 +36,6 @@
     <string name="setting">Setting</string>
     <string name="approve_request">Approve Request</string>
     <string name="share_book">Share Book</string>
+    <string name="personal_infor">Personal Information</string>
```
Comment (trainer-android-fr) [inline app/src/main/res/values/strings.xml]: WARNING: [UnusedResources] The resource `R.string.personal_infor` appears to be unused
 Unused resources make applications larger and slow down builds.

File: app/src/main/res/values-vi/strings.xml [hunk 4]
```diff
@@ -12,4 +12,6 @@
     <string name="setting">Cài đặt</string>
     <string name="approve_request">Phê duyệt yêu cầu</string>
     <string name="share_book">Chia sẻ sách</string>
+    <string name="personal_infor">Thông tin cá nhân</string>
```
Comment (trainer-android-fr) [inline app/src/main/res/values-vi/strings.xml]: WARNING: [UnusedResources] The resource `R.string.personal_infor` appears to be unused
 Unused resources make applications larger and slow down builds.

File: app/src/main/res/values/strings.xml [hunk 5]
```diff
@@ -36,4 +36,6 @@
...
+    <string name="category_favorite">Category Favorite</string>
```
Comment (trainer-android-fr) [inline app/src/main/res/values/strings.xml]: WARNING: [UnusedResources] The resource `R.string.category_favorite` appears to be unused
 Unused resources make applications larger and slow down builds.

File: app/src/main/res/values-vi/strings.xml [hunk 6]
```diff
@@ -12,4 +12,6 @@
...
+    <string name="category_favorite">Danh mục yêu thích</string>
```
Comment (trainer-android-fr) [inline app/src/main/res/values-vi/strings.xml]: WARNING: [UnusedResources] The resource `R.string.category_favorite` appears to be unused
 Unused resources make applications larger and slow down builds.

File: app/src/main/res/values/strings.xml (hunk 3 above)
Comment (Tuanlvt) [reply app/src/main/res/values/strings.xml]: have used

File: app/src/main/res/values-vi/strings.xml (hunk 4 above)
Comment (Tuanlvt) [reply app/src/main/res/values-vi/strings.xml]: have used

File: app/src/main/res/values/strings.xml (hunk 5 above)
Comment (Tuanlvt) [reply app/src/main/res/values/strings.xml]: have used

File: app/src/main/res/values-vi/strings.xml (hunk 6 above)
Comment (Tuanlvt) [reply app/src/main/res/values-vi/strings.xml]: have used

File: app/src/main/res/layout/fragment_menu_profile_container.xml (hunk 2 above)
Comment (Tuanlvt) [reply app/src/main/res/layout/fragment_menu_profile_container.xml]: will use later

File: app/src/main/java/com/framgia/fbook/screen/menuprofile/MenuProfileContainerFragment.kt [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+  companion object {
+
+    val TAG: String = "MenuProfileContainerFragment"
+    val PROFILE: Int = 0
+    val CATEGORY_FAVORITE: Int = 1
+
+    fun newInstance(position: Int): MenuProfileContainerFragment {
+      var menuProfileContainerFragment: MenuProfileContainerFragment = MenuProfileContainerFragment()
+      var bundle: Bundle = Bundle()
+      bundle.putInt(TAG, position)
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/fbook/screen/menuprofile/MenuProfileContainerFragment.kt]: remove ``TAG`` and define another constant variable

File: app/src/main/java/com/framgia/fbook/screen/menuprofile/MenuProfileContainerFragment.kt [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+      menuProfileContainerFragment.arguments = bundle
+      return menuProfileContainerFragment
+    }
+  }
+
+  private val mNavigator: Navigator by lazy { Navigator(this) }
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/fbook/screen/menuprofile/MenuProfileContainerFragment.kt]: Pls declare about ``lazy`` when u use it in here

File: app/src/main/java/com/framgia/fbook/screen/menuprofile/MenuProfileModule.kt [hunk 9]
```diff
@@ -22,6 +23,13 @@ class MenuProfileModule(private val fragment: Fragment) {
   @FragmentScope
   @Provides
   fun provideNavigator(): Navigator {
-    return Navigator(fragment)
+    return Navigator(mFragment)
+  }
+
+  @FragmentScope
+  @Provides
+  fun provideAdapter(): MenuProfileAdapter {
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/fbook/screen/menuprofile/MenuProfileModule.kt]: pls rename to ``provideMenuProfileAdapter``

File: app/src/main/java/com/framgia/fbook/screen/menuprofile/MenuProfileContainerFragment.kt (hunk 8 above)
Comment (daolq3012) [reply app/src/main/java/com/framgia/fbook/screen/menuprofile/MenuProfileContainerFragment.kt]: Comment in here!

Comment (Tuanlvt) [reply app/src/main/java/com/framgia/fbook/screen/menuprofile/MenuProfileContainerFragment.kt]: để gọn code lại với đẹp code hơn a

Comment (daolq3012) [reply app/src/main/java/com/framgia/fbook/screen/menuprofile/MenuProfileContainerFragment.kt]: cụ thể đi em, tại sao không là 
```kt
private val mNavigator: Navigator = Navigator(this)
```

**Pull request merged**